core/src/main/java/org/apache/iceberg/util/BinPacking.java

```diff
@@ -36,51 +36,59 @@
   public static class ListPacker<T> {
     private final long targetWeight;
     private final int lookback;
+    private final int itemsPerBin;
     private final boolean largestBinFirst;
 
     public ListPacker(long targetWeight, int lookback, boolean largestBinFirst) {
+      this(targetWeight, lookback, largestBinFirst, Integer.MAX_VALUE);
+    }
+
+    public ListPacker(long targetWeight, int lookback, boolean largestBinFirst, int itemsPerBin) {
       this.targetWeight = targetWeight;
       this.lookback = lookback;
+      this.itemsPerBin = itemsPerBin;
```

Comment:
Nit: might want to move this one below `largestBinFirst` so they're in the same order as the parameters list.